pyproject.toml

```diff
@@ -58,14 +58,64 @@ Source = "https://github.com/encode/uvicorn"
 path = "uvicorn/__init__.py"
 
 [tool.hatch.build.targets.sdist]
-include = [
-    "/uvicorn",
-]
+include = ["/uvicorn"]
 
 [tool.ruff]
 select = ["E", "F", "I"]
 ignore = ["B904", "B028"]
-line-length = 120
 
 [tool.ruff.isort]
 combine-as-imports = true
+
+[tool.mypy]
+disallow_untyped_defs = true
+ignore_missing_imports = true
+follow_imports = "silent"
+# NOTE: If you are seeing this, feel free to create a PR to cover the below files.
+exclude = """(?x)
+  ^(
+    |tests/protocols/test_utils.py
+  )$
+"""
+
+[[tool.mypy.overrides]]
+module = "tests.*"
+disallow_untyped_defs = false
+check_untyped_defs = true
+
+[tool.pytest.ini_options]
+addopts = "-rxXs --strict-config --strict-markers"
+xfail_strict = true
+filterwarnings = [
+    "error",
+    'ignore: \"watchgod\" is deprecated\, you should switch to watchfiles \(`pip install watchfiles`\)\.:DeprecationWarning',
+    "ignore:Uvicorn's native WSGI implementation is deprecated.*:DeprecationWarning",
+    "ignore: 'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning"
+]
+
+[tool.coverage.run]
+source_pkgs = ["uvicorn", "tests"]
```

Comment:
The drop on coverage is because I replaced the previous `omit` and `include` by `source_pkgs`, like `starlette`.